src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFill.cs

```diff
@@ -99,12 +99,10 @@ private static bool IsDefaultValue(IArgumentOperation argumentOperation)
                 // but since VB doesn't properly support Span, just don't consider it
                 switch (constantOpt.Value)
                 {
-#pragma warning disable IDE0004 // Remove Unnecessary Cast - false positive
-                    case (byte)0 or (short)0 or (int)0 or (long)0 or
+                    case (byte)0 or (short)0 or 0 or (long)0 or
                         (sbyte)0 or (ushort)0 or (uint)0 or (ulong)0 or
                         false or '\0':
                         return true;
-#pragma warning restore IDE0004 // Remove Unnecessary Cast
```

Comment:
FYI @huoyaoyuan @buyaa-n

Reply:
Using 0U, 0L and 0UL may look better since there's no cast on int. Just stylish.

Reply:
We cannot use literal suffix for all numeric types, so the cast will remain in some and we'll end up with 3 styles:

- the plain literal (for `int`)
- the literal with suffix (for `long`, etc.)
- the cast (for `byte`, etc.)

I think it's less confusing to avoid suffix if the other styles will remain existing anyway.

Alternatively, I can keep the suppression and have the cast on all types.

Reply:
I disabled auto merge until this conversation is resolved.

Reply:
@huoyaoyuan Let me know if you still prefer using literal suffix, or if you think the current approach is good enough, or if you prefer keeping the cast and suppression.

Reply:
I'm really OK with either approach. For preference, I'm against suppression for stylish warnings.
I treat plain literal and literal with suffix as similar styles. This is my personal preference and we should follow the repo preference, if any.

Reply:
> we should follow the repo preference, if any.

@mavasani Are there any coding guidelines regarding cast vs literal suffix? If not, I can push a commit to apply @huoyaoyuan's original suggestion.

Reply:
Not really, I am fine with either approach here as well.

Reply:
@huoyaoyuan Can you take a look at the last commit? Thanks!